lexos/models/stats_model.py

```diff
@@ -64,109 +68,197 @@ def _id_temp_label_map(self) -> IdTempLabelMap:
+        return self._doc_term_matrix.iloc[self._stats_option.active_file_ids]
+
+    @property
+    def _token_type_str(self) -> str:
+        """:return: the token type that was used when calculating the stats."""
+        if self._test_id_temp_label_map is not None:
+            return self._test_token_type_str
+        else:
+            # Get dtm front end options.
+            dtm_options = MatrixReceiver().options_from_front_end()
+            # Get the correct current type.
+            token_type = dtm_options.token_option.token_type
+            return "terms" if token_type == "word" else "characters"
+
+    def get_corpus_stats(self) -> CorpusStats:
         """Converts word lists completely to statistic.
 
         :return: a typed tuple that holds all statistic of the entire corpus.
         """
-
         # Check if empty corpus is given.
-        assert np.sum(self._doc_term_matrix.values) > 0, EMPTY_LIST_MESSAGE
+        assert not self._active_doc_term_matrix.empty, EMPTY_DTM_MESSAGE
 
-        # Initialize two dictionaries to hold anomaly analysis result.
-        file_anomaly_iqr = {}
-        file_anomaly_std_err = {}
-        # Get file names.
-        labels = [self._id_temp_label_map[file_id]
-                  for file_id in self._doc_term_matrix.index.values]
-        # Find size of each file.
-        file_sizes = np.sum(self._doc_term_matrix.values, axis=1)
-        # Find average size of all files.
-        average_file_size = round(np.average(file_sizes), 3)
+        # Get the active file ids.
+        active_file_ids = self._active_doc_term_matrix.index
+
+        # Get the file count sums by sum the column.
+        file_sizes = self._active_doc_term_matrix.sum(axis="columns")
+        # Get the average file word counts.
+        mean = file_sizes.mean(axis="index")
+
+        # Get the standard deviation of the file word counts.
+        std_deviation = file_sizes.std(axis="index")
+
+        # Get the iqr of the file word counts.
+        first_quartile = file_sizes.quantile(0.25)
+        third_quartile = file_sizes.quantile(0.75)
+        iqr = third_quartile - first_quartile
 
         # Standard error analysis: assume file sizes are normally distributed;
         # we detect anomaly by finding files with sizes that are more than two
         # standard deviation away from the mean. In another word, we find files
         # with sizes that are not in the major 95% range.
-        # Find the standard deviation.
-        std_dev_file_size = np.std(file_sizes).item()
-        # Find file anomaly.
-        for count, label in enumerate(labels):
-            if file_sizes[count] > average_file_size + 2 * std_dev_file_size:
-                file_anomaly_std_err.update({label: 'large'})
-            elif file_sizes[count] < average_file_size - 2 * std_dev_file_size:
-                file_anomaly_std_err.update({label: 'small'})
+        anomaly_se_small = list(set(
```

Comment:
I had to force this to be a list since the Jsonify function can't seem to deal with set.

Reply:
You can just use list comprehension.